src/Access/UsersConfigAccessStorage.cpp

```diff
@@ -118,11 +118,15 @@ namespace
         const std::unordered_set<UUID> & allowed_profile_ids,
         const std::unordered_set<UUID> & allowed_role_ids,
         bool allow_no_password,
-        bool allow_plaintext_password)
+        bool allow_plaintext_password,
+        bool escape_dot_in_user_name)
     {
         const bool validate = true;
         auto user = std::make_shared<User>();
-        user->setName(user_name);
+        String unescaped_user_name = user_name;
+        if (!escape_dot_in_user_name)
+            Poco::replaceInPlace(unescaped_user_name, "\\.", ".");
```

Comment:
Can we simply do this always, without any options? This looks safe for users.

Reply:
Oh, you already raised the question

>So I wonder if the setting is needed or I should make this behaviour the default and non-changeable.

Yes, let's remove the configuration directive.